packages/@icon-magic/config-reader/src/schemas/config-schema.ts

```diff
@@ -3,14 +3,61 @@ const variantProperties = {
   path: { type: 'string', minLength: 1 }
 };
 
+const pluginProperties = {
+  name: { type: 'string' },
+  iterants: {
+    type: ['array'],
+    items: {
+      type: 'string'
+    }
+  },
+  params: ['object'],
+  writeToOutput: {
+    type: 'boolean',
+    default: true
+  }
+};
+
 const buildConfigProperties = {
-  outputPath: { type: ['string', 'null'] },
-  plugins: { type: ['array', 'null'] }
+  outputPath: {
+    type: ['string', 'null'],
+    minLength: 1
+  },
+  plugins: {
+    type: ['array', 'null'],
+    items: {
+      type: ['object'],
+      properties: pluginProperties,
+      additionalProperties: true
```

Comment:
What are some of the additionalProperties here?

Reply:
removed